internal/fieldmask/fieldmask.go

```diff
@@ -0,0 +1,165 @@
+package fieldmask
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/tidwall/gjson"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+)
+
+const key = "fields"
+
+// FieldMask is recursive structure to define a path mask
+type FieldMask map[string]FieldMask
+
+// New create a FieldMask from the input array of paths. The array should contain JSON paths with dit "." notation.
+func New(paths []string) FieldMask {
+	mask := make(FieldMask)
+	for _, path := range paths {
+		current := mask
+		fields := strings.Split(path, ".")
+		for _, field := range fields {
+			c, ok := current[field]
+			if !ok {
+				c = make(FieldMask)
+				current[field] = c
+			}
+			current = c
+		}
+	}
+	return mask
+}
+
+// Filter takes a Proto message as input and updates the message according to the FieldMask.
+func (fm FieldMask) Filter(message proto.Message) {
+	if len(fm) == 0 {
+		return
+	}
+
+	reflect := message.ProtoReflect()
+	reflect.Range(func(fd protoreflect.FieldDescriptor, v protoreflect.Value) bool {
+		mask, ok := fm[string(fd.Name())]
+		if !ok {
+			reflect.Clear(fd)
```

Comment:
the godoc for `Clear` says it is not safe for concurrent use

are we 100% sure are path to here will not be concurrent?  I'm guessing it is now, assuming GRPC manages individual messages simlarly, but I'm still new to GRPC, so don't what to assume.

Reply:
The filtering happens in the response proto. There can be only one response to a request, handled by a single goroutine. If in case we are modifying the response in another Interceptor, still I guess it won't be concurrent access because the interceptors are chained.

Reply:
those clarifications help thanks